src/shared/components/query/CancerStudySelector.tsx

```diff
@@ -307,45 +313,104 @@ export default class CancerStudySelector extends React.Component<
+                                                                          .length
+                                                                          ? 'matching filter'
+                                                                          : ''
+                                                                  }  (${
+                                                                      shownStudies.length
+                                                                  })`}
+                                                        </strong>
+                                                    </label>
+                                                </Then>
+                                                <Else>
+                                                    <label>
+                                                        <input
+                                                            type="checkbox"
+                                                            data-test="selectAuthorizedStudies"
+                                                            style={{ top: -2 }}
+                                                            onClick={
+                                                                this.handlers
+                                                                    .onCheckAuthorizedFiltered
+                                                            }
+                                                            checked={
+                                                                shownAndSelectedStudies.length ===
+                                                                    shownAndAuthorizedStudies.length &&
+                                                                shownAndAuthorizedStudies.length >
+                                                                    0
+                                                            }
+                                                        />
+                                                        <strong>
+                                                            {shownAndSelectedStudies.length ===
+                                                                shownAndAuthorizedStudies.length &&
+                                                            shownAndAuthorizedStudies.length >
+                                                                0
+                                                                ? `Deselect all authorized studies ${
+                                                                    shownStudies.length <
+                                                                      this.store
+                                                                          .cancerStudies
+                                                                          .result
+                                                                          .length
```

Comment:
Just wanted to make sure that we are waiting for this data-fetching process to finish somewhere, right? I can't see it's waiting for it in this component.

Reply:
default value exists for this.store.cancerStudies.